webroot/js/app-video-only.js

```diff
@@ -0,0 +1,265 @@
+import { h, Component } from 'https://unpkg.com/preact?module';
+import htm from 'https://unpkg.com/htm?module';
+const html = htm.bind(h);
+
+import { OwncastPlayer } from './components/player.js';
+
+import {
+  addNewlines,
+  pluralize,
+} from './utils/helpers.js';
+import {
+  URL_CONFIG,
+  URL_STATUS,
+  TIMER_STATUS_UPDATE,
+  TIMER_STREAM_DURATION_COUNTER,
+  TEMP_IMAGE,
+  MESSAGE_OFFLINE,
+  MESSAGE_ONLINE,
+} from './utils/constants.js';
+
+export default class VideoOnly extends Component {
+  constructor(props, context) {
+    super(props, context);
+
+    this.state = {
+      configData: {},
+
+      playerActive: false, // player object is active
+      streamOnline: false,  // stream is active/online
+
+      //status
+      streamStatusMessage: MESSAGE_OFFLINE,
+      viewerCount: '',
+      sessionMaxViewerCount: '',
+      overallMaxViewerCount: '',
+    };
+
+    // timers
+    this.playerRestartTimer = null;
+    this.offlineTimer = null;
+    this.statusTimer = null;
+    this.streamDurationTimer = null;
+
+    this.handleOfflineMode = this.handleOfflineMode.bind(this);
+    this.handleOnlineMode = this.handleOnlineMode.bind(this);
+
+    // player events
+    this.handlePlayerReady = this.handlePlayerReady.bind(this);
+    this.handlePlayerPlaying = this.handlePlayerPlaying.bind(this);
+    this.handlePlayerEnded = this.handlePlayerEnded.bind(this);
+    this.handlePlayerError = this.handlePlayerError.bind(this);
+
+    // fetch events
+    this.getConfig = this.getConfig.bind(this);
+    this.getStreamStatus = this.getStreamStatus.bind(this);
+  }
+
+  componentDidMount() {
+    this.getConfig();
+
+    this.player = new OwncastPlayer();
+    this.player.setupPlayerCallbacks({
+      onReady: this.handlePlayerReady,
+      onPlaying: this.handlePlayerPlaying,
+      onEnded: this.handlePlayerEnded,
+      onError: this.handlePlayerError,
+    });
+    this.player.init();
+  }
+
+  componentWillUnmount() {
+    // clear all the timers
+    clearInterval(this.playerRestartTimer);
+    clearInterval(this.offlineTimer);
+    clearInterval(this.statusTimer);
+    clearInterval(this.streamDurationTimer);
+  }
+
+  // fetch /config data
+  getConfig() {
```

Comment:
nit: I tend to prefer the `async` and `await` syntax when using fetch, just since it flattens out the callbacks. But I'll defer to what you like to do!